bitcoin_client_js/src/lib/appClient.ts

```diff
@@ -402,12 +403,69 @@ export class AppClient {
     return result.toString('base64');
   }
 
+  /* Performs any additional check on the genetated address before returning it.*/
+  private async validateAddress(
+    address: string,
+    walletPolicy: WalletPolicy,
+    change: number,
+    addressIndex: number
+  ) {
+    if (change !== 0 && change !== 1)
+      throw new Error('Change can only be 0 or 1');
+    if (addressIndex < 0 || !Number.isInteger(addressIndex))
+        throw new Error('Invalid address index');
+    const appAndVer = await this.getAppAndVersion();
+    let network;
+    if (appAndVer.name === 'Bitcoin Test') {
+      network = networks.testnet;
+    } else if (appAndVer.name === 'Bitcoin') {
+      network = networks.bitcoin;
+    } else {
+      throw new Error(
+        `Invalid network: ${appAndVer.name}. Expected 'Bitcoin Test' or 'Bitcoin'.`
+      );
+    }
+    let expression = walletPolicy.descriptorTemplate;
+    for (let i = 0; i < walletPolicy.keys.length; i++) {
+      const keyPath = walletPolicy.keys[i] + '/' + change + '/' + addressIndex;
+      expression = expression
+        .replace('@' + i + '/**', keyPath)
+        .replace('@' + i + '/<0;1>', keyPath);
```

Comment:
This would not work for policies containing '<M;N>' different than '<0;1>', which are supported.
`/<M;N>' should be replaced with `/M` if `change == 0`. with `/N` otherwise.

I should probably have typed the `change` parameter as `boolean` − this is my mistake (probably because wallet policies weren't clearly defined yet when I wrote the initial clients). Will think how to fix it for a future release.